compiler/rustc_interface/src/util.rs

```diff
@@ -212,27 +212,27 @@ pub fn setup_callbacks_and_run_in_thread_pool_with_globals<F: FnOnce() -> R + Se
 }
 
 fn load_backend_from_dylib(path: &Path) -> fn() -> Box<dyn CodegenBackend> {
-    let lib = DynamicLibrary::open(path).unwrap_or_else(|err| {
+    let lib = unsafe { Library::new(path) }.unwrap_or_else(|err| {
         let err = format!("couldn't load codegen backend {:?}: {:?}", path, err);
         early_error(ErrorOutputType::default(), &err);
     });
-    unsafe {
-        match lib.symbol("__rustc_codegen_backend") {
-            Ok(f) => {
-                mem::forget(lib);
-                mem::transmute::<*mut u8, _>(f)
-            }
-            Err(e) => {
+
+    let backend_sym =
+        unsafe { lib.get::<fn() -> Box<dyn CodegenBackend>>(b"__rustc_codegen_backend") }
```

Comment:
I had a little trouble parsing this one. Can you create a type definition for it?

Reply:
Done.